nemo_automodel/components/models/deepseek_v3/state_dict_adapter.py

```diff
@@ -74,13 +79,20 @@ def _add_quantization_scale_inv_tensors(self, state_dict: dict[str, Any]) -> dic
             if key.endswith(".weight") and not any(
                 non_quantized_key in key for non_quantized_key in non_quantized_keys
             ):
+                value = value.to(dtype=torch.float8_e4m3fn)
+                state_dict[key] = value
                 expected_scale_shape = calculate_scale_shape(value)
-                weight_scale_inv_state_dict[key + "_scale_inv"] = torch.ones(expected_scale_shape, dtype=self.dtype)
+                # Create scale_inv on the same device as the weight to avoid device mismatch during dequantization
+                weight_scale_inv_state_dict[key + "_scale_inv"] = torch.ones(
+                    expected_scale_shape, dtype=torch.float32, device=value.device
+                )
```

Comment:
will FSDP's mixed precision policy cause issues for parameters like this? e.g. if MP policy is BF16 and these tensors are intended to be in fp8/fp32, what happens?